.infrastructure/helm-chart/todoapp/charts/mysql/templates/statefulSet.yml

```diff
@@ -0,0 +1,89 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{.Chart.Name}}
+  namespace: {{.Values.namespace}}
+spec:
+  selector:
+    matchLabels:
+      app: {{.Chart.Name}}
+  replicas: {{.Values.mysql.statefulSet.replicas}}
+  serviceName: {{.Chart.Name}}
+  template:
+    metadata:
+      labels:
+        app: {{.Chart.Name}}
+    spec:
+      containers:
+      - name: {{.Chart.Name}}
+        image: {{.Values.mysql.image.repository}}:{{.Values.mysql.image.tag}}
+        env:
+        {{- range $k, $v := .Values.mysql.secrets}}
+        - name: {{$k}}
+          valueFrom:
+            secretKeyRef:
+              name: mysql-secrets
+              key: {{$k}}
+        {{- end}}
+        - name: MYSQL_DATABASE
+          value: app_db
+        ports:
+        - name: {{.Chart.Name}}
+          containerPort: 3306
+        volumeMounts:
+        - name: data
+          mountPath: /var/lib/mysql
+          subPath: {{.Chart.Name}}
+        - name: config-map
+          mountPath: /docker-entrypoint-initdb.d
+        resources:
+          requests:
+            cpu: {{.Values.mysql.resources.cpu.request}}
+            memory: {{.Values.mysql.resources.memory.request}}
```

Comment:
Ensure that the `mysql.resources.cpu.request` and `mysql.resources.memory.request` values are defined in your `values.yaml` file. If they are not defined, this could lead to issues when deploying the chart. Consider providing default values or handling the case where they might be undefined.